src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,380 @@
+    }
+
+    /**
+     * @notice Sets `who` address as the recipient.
+     * @param who Recipient address.
+     * @dev `who` address should have been whitelisted using `kiss`.
+     */
+    function pick(address who) external isMate {
+        require(bud[who] == 1 || who == address(0), "RwaOutputConduit3/not-bud");
+        to = who;
+        emit Pick(who);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Method to swap DAI contract balance to GEM through PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     */
+    function push() external isMate {
+        _doPush(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     * @param wad DAI amount.
+     */
+    function push(uint256 wad) external isMate {
+        _doPush(wad);
+    }
+
+    /**
+     * @notice Flushes out any DAI balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external isMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `usr` address.
+     * @param usr Destination address
+     * @dev Can be called only by admin.
+     */
+    function quitGem(address usr) external auth {
+        gem.transfer(usr, gem.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Calculate amount of GEM received for swapping DAI through PSM.
+     * @param wad DAI amount.
+     */
+    function daiToGem(uint256 wad) public view returns (uint256) {
+        return mul(wad, WAD) / add(WAD, psm.tout()) / to18ConvertionFactor;
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @param wad DAI amount.
+     */
+    function _doPush(uint256 wad) internal {
+        require(to != address(0), "RwaOutputConduit3/to-not-picked");
+
+        // We might lose some dust here because of rounding errors. I.e.: USDC has 6 dec and DAI has 18.
+        uint256 gemAmount = daiToGem(wad);
+        require(gemAmount > 0, "RwaOutputConduit3/insufficient-swap-gem-amount");
+
+        uint256 prevGemBalance = gem.balanceOf(address(this));
+
+        psm.buyGem(address(this), gemAmount);
+
+        uint256 gemBalance = gem.balanceOf(address(this));
+        uint256 gemPushAmt = sub(gemBalance, prevGemBalance);
```

Comment:
Consider adding a require that the balance diff is gemAmount and just using that instead of gemPushAmt. Otherwise it's weird to old both variables. Also, it's better gas-wise to use gem.balanceOf inline here. Also try to refrain from using both `Amt` and `Amount` suffixes.

Reply:
Do we really even need the require. I think we should just try to `push` `gemAmount` as the psm is an internal contract that won't cheat us.
There is not way to not receive that amount: https://github.com/makerdao/dss-psm/blob/master/src/psm.sol#L128

Reply:
done